cockatrice/src/abstractcarditem.cpp

```diff
@@ -139,7 +139,7 @@ void AbstractCardItem::paintPicture(QPainter *painter, const QSizeF &translatedS
         painter->setBackgroundMode(Qt::OpaqueMode);
         QString nameStr;
         if (facedown)
-            nameStr = "# " + QString::number(id);
+            nameStr = "# " + (obscured ? "???" : QString::number(id));
```

Comment:
hiding this client side does not fix the issue, a user could modify their client and remove this in order to gain advantage, the solution I propose is to create new ids server side for each card, a new id is already created every time a card is moved to play face down.